From: Lidong Zhong <lidong.zhong@suse.com>
Date: Fri, 21 Apr 2017 15:21:38 +0800
Subject: [PATCH] md/raid1/10: remove unused queue
Git-commit: 296617581eac713b3fda588216ae6d16d1e76dd5
Git-repo: git://git.kernel.org/pub/scm/linux/kernel/git/shli/md.git
Patch-mainline: Queued in subsystem maintainer repository
References: FATE#321488

A queue is declared and get from the disk of the array, but it's not
used anywhere. So removing it from the source.

Signed-off-by: Lidong Zhong <lzhong@suse.com>
Acted-by: Guoqing Jiang <gqjiang@suse.com>
Signed-off-by: Shaohua Li <shli@fb.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 drivers/md/raid1.c  | 3 ---
 drivers/md/raid10.c | 2 --
 2 files changed, 5 deletions(-)

diff --git a/drivers/md/raid1.c b/drivers/md/raid1.c
index 57611f4..14a9d36 100644
--- a/drivers/md/raid1.c
+++ b/drivers/md/raid1.c
@@ -2961,7 +2961,6 @@ static struct r1conf *setup_conf(struct mddev *mddev)
 	err = -EINVAL;
 	spin_lock_init(&conf->device_lock);
 	rdev_for_each(rdev, mddev) {
-		struct request_queue *q;
 		int disk_idx = rdev->raid_disk;
 		if (disk_idx >= mddev->raid_disks
 		    || disk_idx < 0)
@@ -2974,8 +2973,6 @@ static struct r1conf *setup_conf(struct mddev *mddev)
 		if (disk->rdev)
 			goto abort;
 		disk->rdev = rdev;
-		q = bdev_get_queue(rdev->bdev);
-
 		disk->head_position = 0;
 		disk->seq_start = MaxSector;
 	}
diff --git a/drivers/md/raid10.c b/drivers/md/raid10.c
index acb3f46..5de951b 100644
--- a/drivers/md/raid10.c
+++ b/drivers/md/raid10.c
@@ -3644,7 +3644,6 @@ static int raid10_run(struct mddev *mddev)
 
 	rdev_for_each(rdev, mddev) {
 		long long diff;
-		struct request_queue *q;
 
 		disk_idx = rdev->raid_disk;
 		if (disk_idx < 0)
@@ -3663,7 +3662,6 @@ static int raid10_run(struct mddev *mddev)
 				goto out_free_conf;
 			disk->rdev = rdev;
 		}
-		q = bdev_get_queue(rdev->bdev);
 		diff = (rdev->new_data_offset - rdev->data_offset);
 		if (!mddev->reshape_backwards)
 			diff = -diff;
-- 
2.10.2

